backend/data/src/main/kotlin/io/tolgee/service/key/ResolvingKeyImporter.kt

```diff
@@ -116,6 +125,28 @@ class ResolvingKeyImporter(
     return result
   }
 
+  private fun addOrUpdateTranslation(
+    key: Key,
+    language: LanguageDto,
+    translationText: String,
+    addToList: (TranslationToModify) -> Boolean,
+    new: Boolean,
+    existingTranslation: Translation?,
+  ) {
+    if (new) {
+      val translation =
+        Translation(translationText).apply {
+          this.key = key
+          this.language = entityManager.getReference(Language::class.java, language.id)
+        }
+      addToList(TranslationToModify(translation, translationText, true))
+    } else {
+      if (existingTranslation != null) {
```

Comment:
This condition feels weird, as you are checking if it's new in the condition before, so the else statement suggests that existing translation should exist but you are still checking it.

Reply:
I did this because `TranslationToModify` needs a non-nullable existingTranslation, so if I don't check this on null, the compiler will yell at me